internal/appsec/listener/grpcsec/grpc.go

```diff
@@ -100,35 +103,36 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
 		return
 	}
 
-	// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
-	// see if the associated user should be blocked. Since we don't control the execution flow in this case
-	// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
-	dyngo.On(op, func(userIDOp *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
-		values := make(map[string]any, 1)
-		for addr := range l.addresses {
-			if addr == UserIDAddr {
-				values[UserIDAddr] = args.UserID
+	// Listen to the UserID address if the WAF rules are using it
+	if l.isSecAddressListened(UserIDAddr) {
```

Comment:
Not gonna say anything, but this change seems unrelated to the passlist support 🫣

Reply:
Discussed at the standup: let's start relying on tools such as git machete to make it simpler to have separate refactoring PR stacks